src/System.Linq/src/System/Linq/ToCollection.cs

```diff
@@ -30,6 +30,22 @@ public static List<TSource> ToList<TSource>(this IEnumerable<TSource> source)
             return source is IIListProvider<TSource> listProvider ? listProvider.ToList() : new List<TSource>(source);
         }
 
+        public static List<TSource> ToList<TSource>(this IEnumerable<TSource> source, int capacity)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), "Non-negative number required.");
+            }
+
+            var list = new List<TSource>(capacity);
+            list.AddRange(source);
```

Comment:
This ignores the several optimised versions available through `IIListProvider`, so `var list = source.ToList(); if (list.Capacity < capacity) list.Capacity = capacity;` might win on balance because it has (possibly) one more resize than this way but skips n*2 or more virtual method calls.